react/actions/actionCreators.js

```diff
@@ -12,13 +12,31 @@ export default {
     // UI controls
 }
 
+function injectScreenshots(response, callback){
+  var gists = response.hits.hits;
+  var gistsWithoutThumb = gists.filter(function(d){ return !d._source.thumb; })
+  var gistsWithoutThumbIds = gistsWithoutThumb.map(function(d, i){ return d._id; });;
+  if(gistsWithoutThumbIds.length > 0){
+    d3.json("http://69.164.216.89:8080")
+      .post(JSON.stringify({"gists":gistsWithoutThumbIds}), function(error, data) {
+        gistsWithoutThumb.forEach(function(d){ return d._source.thumbBase64 = data[d._id]; });
```

Comment:
what happens if the server returns an error or can't connect? will data be defined? we should check first so that the callback always gets called. (I wouldn't want a problem with thumbnails to obscure search results)